src/gf.c

```diff
@@ -134,6 +142,13 @@ static inline int cache_match(jl_value_t **args, size_t n, jl_tupletype_t *sig,
                 // to match exactly either. this is cached as Type{T}.
                 // analogous to the situation with tuples.
             }
+            else if (jl_is_typector(a)) {
+                // in the case where a is exactly a
+                // TypeConstructor, tp0 must be exactly
+                // Type{TypeConstructor}
+                if (tp0 != (jl_value_t*)jl_typector_type)
```

Comment:
`a` is the arg here not its type, so i thought this is testing `if isa(a, TypeConstructor) && slottype.name === Type.name` => `? is(slottype, Type{TypeConstructor})`